custom_components/versatile_thermostat/sensor.py

```diff
@@ -423,4 +424,41 @@ def native_unit_of_measurement(self) -> str | None:
     @property
     def suggested_display_precision(self) -> int | None:
         """Return the suggested number of decimal digits for display."""
-        return 2
+        return 
+    
+    
+class TargetTemperatureSensor(VersatileThermostatBaseEntity, SensorEntity):
+    """Rapresentation of the target temperature in sensor format"""
+
+    def __init__(self, hass: HomeAssistant, unique_id, name, entry_infos) -> None:
+        """Initialize the target temperature sensor"""
+        super().__init__(hass, unique_id, entry_infos.get(CONF_NAME))
+        self._attr_name = "Target Temperature"
+        self._attr_unique_id = f"{self._device_name}_target_temp"
+
+    @callback
+    async def async_my_climate_changed(self, event: Event = None):
+        """Called when my climate have change"""
+        _LOGGER.debug("%s - climate state change", self._attr_unique_id)
+
+        old_state = self._attr_native_value
+        self._attr_native_value = self._target_temp
+        if old_state != self._attr_native_value:
+            self.async_write_ha_state()
+        return
+    
+    @property
+    def icon(self) -> str | None:
+        return "mdi:sun-thermometer-outline"
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        return SensorDeviceClass.TEMPERATURE
+
+    @property
+    def state_class(self) -> SensorStateClass | None:
+        return SensorStateClass.MEASUREMENT
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
```

Comment:
Here you should use the Home Assistant setting of the user. See climate.py :

```
@property
    def temperature_unit(self) -> str:
        """Return the unit of measurement."""
        if self._is_over_climate and self.underlying_entity(0):
            return self.underlying_entity(0).temperature_unit

        return self._unit
```


and:

`self._unit = self._hass.config.units.temperature_unit`